core/domain/opportunity_services.py

```diff
@@ -407,6 +408,214 @@ def update_opportunities_with_new_topic_name(topic_id, topic_name):
+
+    Args:
+        skill_id: str. The skill_id of the opportunity.
+        skill_description: str. The skill_description of the opportunity.
+
+    Raises:
+        Exception: If a SkillOpportunityModel corresponding to the supplied
+            skill_id already exists.
+    """
+    skill_opportunity_model = (
+        opportunity_models.SkillOpportunityModel.get_by_id(skill_id))
+    if skill_opportunity_model is not None:
+        raise Exception(
+            'SkillOpportunity corresponding to skill ID %s already exists.' % (
+                skill_id))
+
+    questions, _, _ = (
+        question_fetchers.get_questions_and_skill_descriptions_by_skill_ids(
+            constants.MAX_QUESTIONS_PER_SKILL, [skill_id], ''))
+    skill_opportunity = opportunity_domain.SkillOpportunity(
+        skill_id=skill_id,
+        skill_description=skill_description,
+        question_count=len(questions)
+    )
+    _save_skill_opportunities([skill_opportunity])
+
+
+def _save_skill_opportunities(skill_opportunities):
+    """Saves SkillOpportunity domain objects into datastore as
+    SkillOpportunityModel objects.
+
+    Args:
+        skill_opportunities: list(SkillOpportunity). A list of SkillOpportunity
+            domain objects.
+    """
+    skill_opportunity_models = []
+    for skill_opportunity in skill_opportunities:
+        skill_opportunity.validate()
+        model = opportunity_models.SkillOpportunityModel(
+            id=skill_opportunity.id,
+            skill_description=skill_opportunity.skill_description,
+            question_count=skill_opportunity.question_count,
+        )
+        skill_opportunity_models.append(model)
+    opportunity_models.SkillOpportunityModel.put_multi(skill_opportunity_models)
+
+
+def update_skill_opportunity_skill_description(skill_id, new_description):
+    """Updates the skill_description of the SkillOpportunityModel with
+    new_description.
+
+    Args:
+        skill_id: str. The corresponding skill_id of the opportunity.
+        new_description: str. The new skill_description.
+    """
+    skill_opportunity = _get_skill_opportunity(skill_id)
+    if skill_opportunity is not None:
+        skill_opportunity.skill_description = new_description
+        _save_skill_opportunities([skill_opportunity])
+
+
+def _get_skill_opportunity(skill_id):
+    """Returns the SkillOpportunity domain object representing a
+    SkillOpportunityModel with the supplied skill_id in the datastore.
+
+    Args:
+        skill_id: str. The corresponding skill_id of the opportunity.
+
+    Returns:
+        SkillOpportunity or None. The domain object representing a
+            SkillOpportunity with the supplied skill_id, or None if it does not
+            exist.
+    """
+    skill_opportunity_model = (
+        opportunity_models.SkillOpportunityModel.get_by_id(skill_id))
+    if skill_opportunity_model is not None:
+        return get_skill_opportunity_from_model(skill_opportunity_model)
+    return None
```

Comment:
The Code-Climate checks are complaining about this line, but I'm not sure of the best way to address it? The method itself is prefixed with an underscore since it's only meant to be used internally in this file, so it doesn't make sense to test this specific code path directly.

Reply:
@sagangwee Maybe you can write a test for `update_skill_opportunity_skill_description` with invalid skill_id and description which will test the None case for `_get_skill_opportunity` method?

Reply:
Okay I'll try that.